Some minor fixes to follow the style guide, fixes #339

Laura-O opened the pull request:
These are some fixes following the style guide.

Specifically:
- begin -> start
- scramble -> scramble sequence
- allowed -> permitted

Closes #339

Laura-O committed:
Some minor fixes to follow the style guide, fixes #339

graned -> granted

viroulep commented on wca-guidelines.md:
```diff
@@ -161,9 +161,9 @@ To be more informative, each Guideline is classified using one of the following
-- A7g+++) [RECOMMENDATION] The WCA Delegate should use stronger discretion when awarding more than one extra attempt to the same new competitor. If more than one extra attempt is awarded, it should be for a different incident.
+- A7g+) [EXAMPLE] Examples of incidents that could grant a new competitor an extra attempt: not waiting for the green light, improperly starting or stopping the timer, going over inspection time due to misunderstanding how to start an attempt, or other procedural time penalties.
+- A7g++) [CLARIFICATION] Individual penalties cannot be waived for new competitors. A full extra attempt must be granted.
+- A7g+++) [RECOMMENDATION] The WCA Delegate should use stronger discretion when granting more than one extra attempt to the same new competitor. If more than one extra attempt is graned, it should be for a different incident.
```
A typo slipped in ! "Graned" instead of granted.

Laura-O replied:
Thanks for checking so carefully. I'm incompetent...

viroulep committed:
Use of recommended words by the style guide.

Standardization pass on the Regulations+Guidelines.
    - Replaced words with their recommended alternative
    - Implemented Lucas' suggestion to simplify things by also using "start" as a noun.

viroulep commented:
I just pushed another standardization pass to this PR.
Mainly, I implemented Lucas' suggestion to also use "start" as a noun instead of "beginning". The result looks fine, let me know what you think.

I also went through all the style guide and substituted the words by their recommended alternative.
There is a bunch of "give" uses left, but I don't think it's appropriate to use "grant" in their context.

Laura-O commented:
I thought about changing these "start" mentions as well, but I didn't because they are rather general descriptions and not concerning the actual solving procedure.
However, that was just a spontaneous decision, I'm completely fine with these changes.
